### ISSUE TITLE
🌟Feature request: Menu grace area for submenus

### ISSUE THREAD

**Issue by yekta** [melt-ui/melt-ui#177]
The grace area for submenus seems bit too small. Here is a mouse path that causes the submenu to close:

![image](https://github.com/melt-ui/melt-ui/assets/40706601/077828e5-d1b0-4860-a7fc-de4dd6269391)

I suggested a grace area like this instead:

![image](https://github.com/melt-ui/melt-ui/assets/40706601/291c8e5a-a726-41e4-a576-3dbf89e8498f)

During the discussion in Discord someone also shared this article:

https://height.app/blog/guide-to-build-context-menus

**Comment by huntabyte** [comment]
What browser are you using? If I move there fast enough then I'm all good in that green area (chromium browser)

**Comment by huntabyte** [comment]
We not only have a grace area, but also a grace period, if you aren't moving to the submenu fast enough, it assumes you aren't going there and closes. Sometime this week I'll add a few different variations and see how it feels, and also consider exposing the default grace period time as well.

**Comment by yekta** [comment]
I’m using Chrome on MacOS. As far as being able to use mouse fast goes, I think I’d rank pretty high compared to non-developers / FPS gamers. The grace area I proposed was without a duration (or a very long one) since some people would be quite significantly slower with their mouse.

I visited the docs once again just now and currently submenu doesn’t close at all once open. I’m not sure if anything changed since yesterday.
